src/sql/reuse.rs

```diff
@@ -94,6 +95,54 @@ where
     }
 }
 
+/// Direct elimination for complex predicates with nested `and` and `or` expressions
+fn complex_predicate_implies(np: &ConditionExpression, ep: &ConditionExpression) -> bool {
+    match *ep {
+        LogicalOp(ref ect) => {
+            match *np {
+                LogicalOp(ref nct) => {
+                    if nct.operator == ect.operator {
+                        return (complex_predicate_implies(&*nct.left, &*ect.left) && complex_predicate_implies(&*nct.right, &*ect.right)) ||
+                                (complex_predicate_implies(&*nct.left, &*ect.right) && complex_predicate_implies(&*nct.right, &*ect.left));
+                    }
```

Comment:
Nice, this captures the case when the predicates in the `OR` condition are flipped :)